lib/redis_scripts.rb

```diff
@@ -57,8 +57,9 @@ def run(name, *args)
       error.message.include?('NOSCRIPT') or
         raise
       sha, value = redis.pipelined do
-        redis.script 'load', script.content
-        redis.evalsha(script.sha, *args)
+        sha = redis.script 'load', script.content
+        result = redis.evalsha(script.sha, *args)
+        return sha, result
```

Comment:
Won't `return` in this context exit you from the `run` method itself? It might not as `pipelined` may take the block and convert it into a lambda before running, but that complexity is hidden, so I would do the following instead:

```suggestion
        [redis.script('load', script.content), redis.evalsha(script.sha, *args)]
```